Dockerfile

```diff
@@ -0,0 +1,93 @@
+# See the README.md for some example docker invocations with this Dockerfile.
+
+# Use Docker official images for Debian; see https://hub.docker.com/_/debian
+FROM debian:bookworm-slim@sha256:90522eeb7e5923ee2b871c639059537b30521272f10ca86fdbbbb2b75a8c40cd
+
+# Use Debian snapshots; see https://snapshot.debian.org/
+ARG DEBIAN_SNAPSHOT="20250523T000000Z"
+COPY <<EOF /etc/apt/sources.list.d/debian.sources
+Types: deb
+URIs: http://snapshot.debian.org/archive/debian/${DEBIAN_SNAPSHOT}
+Suites: bookworm bookworm-updates
+Components: main
+Signed-By: /usr/share/keyrings/debian-archive-keyring.gpg
+Check-Valid-Until: no
+
+Types: deb
+URIs: http://snapshot.debian.org/archive/debian-security/${DEBIAN_SNAPSHOT}
+Suites: bookworm-security
+Components: main
+Signed-By: /usr/share/keyrings/debian-archive-keyring.gpg
+Check-Valid-Until: no
+EOF
+
+# Install Debian packages:
+RUN --network=default <<EOF
+set -eux
+export DEBIAN_FRONTEND=noninteractive
+apt-get update
+apt-get -y install ruby bundler
+apt-get clean
+EOF
+
+# Create the build user and home directory:
+RUN --network=none <<EOF
+set -eux
+useradd --create-home --shell /bin/sh build
+chown build:build /home/build
+EOF
+
+# Switch to the build user, and into their home directory for clear ownership:
+USER build
+WORKDIR /home/build/fusion-site
+
+# Establish a build script for following steps; this may seem odd, but it
+# allows the bundle/jekyll invocations to work whether running with the
+# copied-in files, or files via a volume mount over the workdir. Otherwise,
+# such a volume mount would also remove the installed gems.
+COPY --chmod=755 <<'EOF' /bin/bundlew
+#!/bin/sh
+set -eu
+bundle config set --local gemfile '../Gemfile'
+bundle config set --local path 'bundle'
+case "$1" in
+    jekyll-serve)
+        shift
+        set -x
+        exec bundle exec jekyll serve \
+                --destination ../site \
+                --host 0.0.0.0 \
+                "$@"
+        ;;
+    jekyll-build)
+        shift
+        set -x
+        exec bundle exec jekyll build \
+                --destination ../site \
+                --strict_front_matter \
+                --incremental \
+                "$@"
+        ;;
+    *)
+        set -x
+        exec bundle "$@"
+        ;;
+esac
+EOF
+
+# Install Ruby packages:
+COPY --chown=build:build Gemfile ../Gemfile
+RUN --network=default bundlew install
+
+# Fix Scss file encoding errors; https://github.com/jekyll/jekyll/issues/4268
+ENV LC_ALL="C.UTF-8" LANG="en_US.UTF-8" LANGUAGE="en_US.UTF-8"
+
+# Copy in all site sources; following commands will be rerun for every change:
+COPY --chown=build:build . .
```

Comment:
[Again, Docker dummy here]  This surprised me, I expected some sort of mount so that the input/output files are shared with the outside context.  I'm also not sure what's meant by the "following commands will be rerun" comment.

I'm also confused how this works with `--livereload`?

Reply:
OK, I think I finally understand what's going on here, so I can ask a more focused observation.

It looks like the `exec-jekyll-build` and `exec-jekyll-serve` bits are effectively shortcuts to things that _could_ be done "from the outside" via `docker run`.   And since we rather obviously will want some sort of wrapper on the outside to encapsulate the port- and mount-binding done in the README examples, I feel like it might be more sensible to do that encapsulation in the same place.

Not a big deal, it just feels like one less moving part to work with, and we can of course consider such changes later.